impl-endpoints/nss/run_endpoint.sh

```diff
@@ -50,5 +50,9 @@ else
     echo "Server params: $SERVER_PARAMS"
     echo "Test case: $TESTCASE"
     ECH_KEY=$(<nss_testdata/ech_key_converted)
-    selfserv -a example.com -n example.com -a client-facing.com -n client-facing.com -p "$PORT" -d "$DB_DIR" -X "$ECH_KEY"
+
+    # Interrupt the server after one second. (Rather than serve one connection
+    # and exit, selfserv listens until it is interrupted.)
+    timeout --preserve-status --signal=2 1 \
```

Comment:
This should probably be a parameter with a sane default. (I might go for something longer than 1s.)

Reply:
That makes sense. How about 2s?

Reply:
That seems fine as a default, but it should be configurable, right?

Reply:
Rather than timing out in the endpoint, what about just doing it in the runner? The runner at any rate needs to know the timeout for each testcase, and docker-compose takes a `--timeout` argument.

Reply:
That's a great idea. In fact, I think what we should do is modify boringSSL and cloudflare-go so that they serve connections forever, like rustls and NSS do, and interrupt them with the same mechanism. I'll make this change here and ping you when it's ready.

Reply:
Alrighty, so I learned that `--timeout` doesn't do what we think it does. This controls the time that docker-compose waits until it interrupts the container *after* the command itself killed. For example, if the timeout is 10 seconds, then it will wait up to 10 seconds after you hit Ctrl-C before shutting down. (See https://github.com/docker/compose/issues/5310.)

Still, it seems sensible to have *all* servers run forever and let the script that calls `docker-compose up` decide how long before killing the servers. That's what this PR now does.

Reply:
> Still, it seems sensible to have all servers run forever and let the script that calls docker-compose up decide how long before killing the servers. That's what this PR now does.

Agreed!